src/columns.jl

```diff
@@ -227,10 +227,10 @@ function _colindex(fnames::Union{Tuple, AbstractArray}, col, default=nothing)
         return 0
     elseif isa(col, Tuple)
         return 0
-    elseif isa(col, Pair{Symbol, <:Pair}) # recursive pairs
-        return _colindex(fnames, col[2])
-    elseif isa(col, Pair{<:Any, <:Any})
+    elseif isa(col, Pair{<:Any, <:Function}) # selector => mapfn
```

Comment:
```suggestion
    elseif isa(col, Pair{<:Any, <:Base.Callable}) # selector => mapfn
```

Reply:
From just a cursory glance at this, we could use `Base.Callable` instead of `Function` to allow for callable types